python/taichi/lang/matrix.py

```diff
@@ -410,6 +412,15 @@ def __setitem__(self, indices, item):
         j = 0 if len(indices) == 1 else indices[1]
         self.set_entry(i, j, item)
 
+    def __len__(self):
+        return self.n
```

Comment:
What if `self.m > 1`? Shall we use `self.m * self.n`?

Reply:
Please see `__iter__`, we return `[[2, 3, 4], [4, 5, 6]]` for `list(a)`, so `len(list(a))` is actually `a.n` in this case, let's make `len(a) == len(list(a))`.